File: app/src/main/java/com/scatl/uestcbbs/module/user/adapter/UserVisitorAdapter.java
package com.scatl.uestcbbs.module.user.adapter;

import android.view.View;
import android.widget.ImageView;

import androidx.annotation.NonNull;

import com.bumptech.glide.Glide;
import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;
import com.google.android.material.imageview.ShapeableImageView;
import com.scatl.uestcbbs.R;
import com.scatl.uestcbbs.entity.VisitorsBean;

public class UserVisitorAdapter extends BaseQuickAdapter<VisitorsBean, BaseViewHolder> {

    int mineId;

    public UserVisitorAdapter(int layoutResId, int mineId) {
        super(layoutResId);
        this.mineId = mineId;
    }

    @Override
    protected void convert(@NonNull BaseViewHolder helper, VisitorsBean item) {
        helper.setText(R.id.item_user_visitor_name, item.visitorName)
                .setText(R.id.item_user_visitor_time, item.visitedTime)
                .addOnClickListener(R.id.item_user_visitor_delete);
        Glide.with(mContext).load(item.visitorAvatar).into((ShapeableImageView) helper.getView(R.id.item_user_visitor_icon));
        helper.getView(R.id.item_user_visitor_delete).setVisibility(mineId == item.visitorUid ? View.VISIBLE : View.GONE);
    }
}

File: app/src/main/java/com/scatl/uestcbbs/module/user/adapter/UserVisitorAdapter.kt
package com.scatl.uestcbbs.module.user.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.scatl.uestcbbs.databinding.ItemUserVisitorBinding
import com.scatl.uestcbbs.entity.VisitorsBean
import com.scatl.uestcbbs.helper.PreloadAdapter
import com.scatl.uestcbbs.helper.ViewBindingHolder
import com.scatl.uestcbbs.util.load

/**
 * Created by sca_tl at 2023/6/9 17:39
 */
class UserVisitorAdapter: PreloadAdapter<VisitorsBean, ItemUserVisitorBinding>() {

    var mineId = 0

    override fun getViewBinding(parent: ViewGroup): ItemUserVisitorBinding {
        return ItemUserVisitorBinding.inflate(LayoutInflater.from(context), parent, false)
    }

    override fun onBindViewHolder(holder: ViewBindingHolder<ItemUserVisitorBinding>, position: Int, item: VisitorsBean?) {
        super.onBindViewHolder(holder, position, item)
        if (item == null) {
            return
        }

        holder.binding.avatar.load(item.visitorAvatar)
        holder.binding.time.text = item.visitedTime
        holder.binding.name.text = item.visitedTime
        holder.binding.deleteBtn.visibility = if (mineId == item.visitorUid) View.VISIBLE else View.GONE
    }
}